android-components/components/feature/search/src/main/java/mozilla/components/feature/search/telemetry/Utils.kt

```diff
@@ -11,80 +11,59 @@ private const val SEARCH_TYPE_SAP_FOLLOW_ON = "sap-follow-on"
 private const val SEARCH_TYPE_SAP = "sap"
 private const val SEARCH_TYPE_ORGANIC = "organic"
 private const val CHANNEL_KEY = "channel"
-private val validCodeSet = setOf(
-    "MOZ2", "MOZ4", "MOZ5", "MOZA", "MOZB", "MOZD", "MOZE", "MOZI", "MOZM", "MOZO", "MOZT",
-    "MOZW", "MOZSL01", "MOZSL02", "MOZSL03", "monline_dg", "monline_3_dg", "monline_4_dg",
-    "monline_7_dg", "firefox-a", "firefox-b", "firefox-b-1", "firefox-b-ab", "firefox-b-1-ab",
-    "firefox-b-d", "firefox-b-1-d", "firefox-b-e", "firefox-b-1-e", "firefox-b-m",
-    "firefox-b-1-m", "firefox-b-o", "firefox-b-1-o", "firefox-b-lm", "firefox-b-1-lm",
-    "firefox-b-lg", "firefox-b-huawei-h1611", "firefox-b-is-oem1", "firefox-b-oem1",
-    "firefox-b-oem2", "firefox-b-tinno", "firefox-b-pn-wt", "firefox-b-pn-wt-us", "ubuntu",
-    "ffab", "ffcm", "ffhp", "ffip", "ffit", "ffnt", "ffocus", "ffsb", "fpas",
-    "newext", "1000969a", null,
-)
 private val validChannelSet = setOf("ts")
 
 /**
  * Get a String in a specific format allowing to identify how an ads/search provider was used.
  *
  * @see [TrackKeyInfo.createTrackKey]
  */
+@Suppress("NestedBlockDepth", "ComplexMethod")
 internal fun getTrackKey(
     provider: SearchProviderModel,
     uri: Uri,
     cookies: List<JSONObject>,
 ): String {
+    var type = SEARCH_TYPE_ORGANIC
     val paramSet = uri.queryParameterNames
-    var code: String? = null
-
-    if (provider.codeParam.isNotEmpty()) {
-        code = uri.getQueryParameter(provider.codeParam)
-        if (code.isNullOrEmpty() &&
-            provider.name == "baidu" &&
-            uri.toString().contains("from=")
-        ) {
-            code = uri.toString().substringAfter("from=", "")
-                .substringBefore("/", "")
-        }
-
-        // For Bug 1751920
-        if (!validCodeSet.contains(code)) {
-            code = "other"
-        }
-
-        // Glean doesn't allow code starting with a figure
-        if (code != null && code.isNotEmpty()) {
-            val codeStart = code.first()
-            if (codeStart.isDigit()) {
-                code = "_$code"
+    var code: String? = "none"
```

Comment:
Could we move this mutable declaration of `code` and `type` inside the `if` statement, and use a hardcoded `"none"` and `SEARCH_TYPE_ORGANIC` in the final return? I think it would make it easier to understand the scope in which these variables can change

That is:
```
if (provider.codeParamName.isNotEmpty()) {
        var type = SEARCH_TYPE_ORGANIC
        var code = uri.getQueryParameter(provider.codeParamName)

// at the very bottom of the function
return TrackKeyInfo(provider.telemetryId, SEARCH_TYPE_ORGANIC, "none").createTrackKey()
```